services/filemanager/filemanager.go

```diff
@@ -87,8 +87,27 @@ func (factory *FileManagerFactoryT) New(settings *SettingsT) (FileManager, error
 	return nil, fmt.Errorf("%w: %s", rterror.InvalidServiceProvider, settings.Provider)
 }
 
-// GetProviderConfigForBackupsFromEnv returns the provider config
-func GetProviderConfigForBackupsFromEnv(ctx context.Context) map[string]interface{} {
+// GetBackupStorageConfig returns the backup storage config based on the user settings
+func GetBackupStorageConfig(ctx context.Context) (string, map[string]interface{}) {
+	backendconfig.DefaultBackendConfig.WaitForConfig(ctx)
+	// TODO: Adapt for multi-tenancy
+	workspaceID := backendconfig.DefaultBackendConfig.GetWorkspaceIDForWriteKey("")
+	useRudderStorage := backendconfig.DefaultBackendConfig.GetDataRetentionSettingsForWorkspaceID(workspaceID).UseRudderServerStorage
+
+	if useRudderStorage {
+		return getProviderAndConfigForBackupsFromEnv()
+	} else {
+		return getProviderAndConfigForBackupsFromWorkspaceSettings(backendconfig.DefaultBackendConfig.GetDataRetentionSettingsForWorkspaceID(workspaceID).StorageBucket)
+	}
+}
+
+// getProviderAndConfigForBackupsFromWorkspaceSettings returns the provider config from the workspace settings
+func getProviderAndConfigForBackupsFromWorkspaceSettings(bucket backendconfig.StorageBucket) (string, map[string]interface{}) {
+	return bucket.Provider, bucket.Config
```

Comment:
Are we confident all required parameters will be there and with the appropriate keys?
E.g. @koladilip recently added support for IAM roles in S3 with #2386 and is using the `workspaceID` as a value for `externalId` parameter.

Reply:
Required parameters might be present. But some optional parameters might miss. Do you think enriching on the server make sense?

Reply:
I don't really know, but I believe we shouldn't release this until we can properly support the same features regardless if we are using control-plane or environment-based config